src/Stratis.Bitcoin/P2P/Peer/NetworkPeer.cs

```diff
@@ -477,7 +474,7 @@ public async Task ConnectAsync(CancellationToken cancellation = default(Cancella
         /// <param name="previous">Previous network state of the peer.</param>
         private async Task OnStateChangedAsync(NetworkPeerState previous)
         {
-            bool insideCallback = this.onDisconnectedAsyncContext.Value == null;
+            bool insideCallback = this.onDisconnectedAsyncContext.Value != null;
```

Comment:
hmmmm is this true? how was this missed up to now?

Reply:
yeah, its difficult to say how we collectively missed this 😄. @zeptin, may have just been lucky to spot this while investigating why he had duplicate connections.